backend/src/hatchling/build.py

```diff
@@ -1,7 +1,10 @@
+from __future__ import annotations
+
 import os
+from typing import Any
 
 
-def get_requires_for_build_sdist(config_settings=None):
+def get_requires_for_build_sdist(config_settings: dict[str, Any] | None = None) -> list[dict[str, Any]]:
```

Comment:
do we need `dict[str, Any]` instead of just `dict`? if so wdyta a module for such common types?

Reply:
look like the config's key is str, this is more precise.

Reply:
Then let's add a `backend/src/hatchling/utils/types.py` with `CONFIG: dict[str, Any]`

Reply:
or `STR_DICT`

Reply:
actually nvm 👍